src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -61,14 +61,23 @@ join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeS
     MembershipList = membership_list(State),
     {ok, Membership, OutgoingMessages, State}.
 
-distance(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2}=_State0, Node) ->
-    % A node j joins the system by sending a subscription request to the node s which is closest to it
-    Threshold = 3,
+try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node) ->
     Dict = get(distance_metrics),
     try dict:find(Node, Dict) of
         {ok, Dist} -> Dist
     catch
          error:E -> {"Error computing distance"}
+    end.
+    
+
+distance(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2, distance = Dist}=_State0, Node) ->
+    % A node j joins the system by sending a subscription request to the node s which is closest to it
+    Threshold = 3,
+    try try_distance() of
```

Comment:
I don't understand what this try statement is doing.  You're running ```try_distance()``` and getting the return value and then returning it, but you're not using the value at all.   What are you trying to do?  If you're trying to use the return value you need to do something like this:

```
Thing = try do_something() of 
  ...
```

...which, allows you to use the return value of the statement run inside of the try.